src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -0,0 +1,897 @@
+#include "common.h"
+
+#undef dout_prefix
+#define dout_prefix *_dout << "ceph_dedup_daemon: " \
+                           << __func__ << ": "
+
+ceph::mutex glock = ceph::make_mutex("glock");
+class SampleDedupWorkerThread;
+std::list<SampleDedupWorkerThread> threads;
+bool all_stop = false;
+
+po::options_description make_usage() {
+  po::options_description desc("Usage");
+  desc.add_options()
+    ("help,h", ": produce help message")
+    ("--pool <POOL> --chunk-pool <POOL>",
+     ": perform deduplication on the target pool")
+    ;
+  po::options_description op_desc("Opational arguments");
+  op_desc.add_options()
+    ("chunk-size", po::value<int>(), ": chunk size (byte)")
+    ("chunk-algorithm", po::value<std::string>(), ": <fixed|fastcdc>, set chunk-algorithm")
+    ("fingerprint-algorithm", po::value<std::string>(), ": <sha1|sha256|sha512>, set fingerprint-algorithm")
+    ("chunk-pool", po::value<std::string>(), ": set chunk pool name")
+    ("max-thread", po::value<int>(), ": set max thread")
+    ("report-period", po::value<int>(), ": set report-period")
+    ("pool", po::value<std::string>(), ": set pool name")
+    ("no-snap", ": do not deduplciate snapshotted object")
+    ("chunk-dedup-threshold", po::value<int>(), ": set the threshold for chunk dedup (number of duplication) ")
+    ("sampling-ratio", po::value<int>(), ": set the sampling ratio (percentile)")
+    ("wakeup-period", po::value<int>(), ": set the wakeup period of crawler thread (sec)")
+    ("fpstore-threshold", po::value<size_t>()->default_value(100_M), ": set max size of in-memory fingerprint store (bytes)")
+    ("conf,c", po::value<std::string>(), "read configuration from the given configuration file")
+    ("id,i", po::value<std::string>(), "set ID portion of my name")
+    ("name,n", po::value<std::string>(), "set name")
+    ("cluster", po::value<std::string>(), "set cluster name (default: ceph)")
+    ("setuser", po::value<std::string>(), "set uid to user or uid (and gid to user's gid)")
+    ("setgroup", po::value<std::string>(), "set gid to group or gid")
+    ("version", "show version and quit")
+    (",d", "run in foreground, log to stderr")
+    (",f", "run in foreground, log to usual location")
+    ("debug_ms", po::value<std::string>(), "set message debug level (e.g. 1)")
+    ("run-once", ": do a single iteration for debug")
```

Comment:
Since you're adding support for a normal ceph-style config, use the normal ceph argument parsing as well so that you don't have to duplicate a bunch of these options between the config declaration and here.